File: EspressoApp/src/main/java/com/example/espressoapp/MainActivity.java
package com.example.espressoapp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    Button button;
    EditText editText;
    TextView textView;

    Button homeButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        bindViews();
    }

    private void bindViews() {
        button = findViewById(R.id.button);
        button.setOnClickListener(v -> updateText());
        editText = findViewById(R.id.editText);
        textView = findViewById(R.id.textView);

        homeButton = findViewById(R.id.home_button);
        homeButton.setOnClickListener(v -> navigateToHome());
    }

    private void updateText() {
        String text = editText.getText().toString();
        textView.setText(text);
    }

    private void navigateToHome() {
        Intent intent = new Intent(this, HomeActivity.class);
        startActivity(intent);
    }
}

File: DeepLinkLauncher/src/main/java/ebj/habinyasuyujin/deeplinklauncher/MainActivity.kt
package ebj.habinyasuyujin.deeplinklauncher

import android.content.Intent
import android.net.Uri
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val editText: EditText = findViewById(R.id.editText)
        val button: Button = findViewById(R.id.button)
        button.setOnClickListener {
            val deepLink = editText.text.toString()
            val intent = Intent(Intent.ACTION_VIEW, Uri.parse(deepLink))
            startActivity(intent)
        }
    }
}